subsys/bluetooth/host/conn.c

```diff
@@ -2949,7 +2949,40 @@ void bt_hci_le_df_connection_iq_report(struct net_buf *buf)
 			cb->cte_report_cb(conn, &iq_report);
 		}
 	}
+
+	bt_conn_unref(conn);
 }
 #endif /* CONFIG_BT_DF_CONNECTION_CTE_RX */
 
+#if defined(CONFIG_BT_DF_CONNECTION_CTE_REQ)
+void bt_hci_le_df_cte_req_failed(struct net_buf *buf)
+{
+	struct bt_df_conn_iq_samples_report iq_report;
+	struct bt_conn *conn;
+	struct bt_conn_cb *cb;
+	int err;
+
+	err = hci_df_prepare_conn_cte_req_failed(buf, &iq_report, &conn);
+	if (err) {
+		BT_ERR("Prepare CTE REQ failed IQ report failed %d", err);
+		return;
+	}
+
+	for (cb = callback_list; cb; cb = cb->_next) {
+		if (cb->cte_report_cb) {
+			cb->cte_report_cb(conn, &iq_report);
+		}
+	}
+
+	STRUCT_SECTION_FOREACH(bt_conn_cb, cb)
+	{
+		if (cb->cte_report_cb) {
+			cb->cte_report_cb(conn, &iq_report);
+		}
+	}
```

Comment:
Hmm, `cte_report_cb` doesn't have `iq_report` (or `info`) as `const`, so the receiver of the callbacks may actually change the data in the struct. Is that intentional?

Reply:
It has `const`: https://github.com/zephyrproject-rtos/zephyr/blob/9d346cc2b248d40742f8633f8fc3f7842c44800a/include/bluetooth/bluetooth.h#L1256-L1257

`sync` pointer doesn't have `const`, as it hasn't got it in `state_changed`, `biginfo`...

Reply:
Oh, my bad, I guess the placement of `const` made me miss it :D